staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -139,38 +140,49 @@ type Requirement struct {
 //     of characters. See validateLabelKey for more details.
 //
 // The empty string is a valid value in the input values set.
-func NewRequirement(key string, op selection.Operator, vals []string) (*Requirement, error) {
-	if err := validateLabelKey(key); err != nil {
+// Note: We must keep the exported function return type as error.
+// The returned value may store in an interface variable, returns a struct pointer might cause the nil check no work
+// But this function guarantees to return a *field.Error.
```

Comment:
comments for an exported function normally start from something like "NewRequirement xxx"

Reply:
see L131